java/ql/src/experimental/Security/CWE/CWE-522/InsecureBasicAuth.ql

```diff
@@ -107,6 +106,16 @@ predicate concatHttpString(Expr protocol, Expr host) {
         .(CompileTimeConstantExpr)
         .getStringValue()
         .regexpMatch("(?i)http(://)?")
+  ) and // Not empty host string
+  (
+    host.(CompileTimeConstantExpr).getStringValue().length() > 0 or
+    host
+        .(VarAccess)
+        .getVariable()
+        .getAnAssignedValue()
+        .(CompileTimeConstantExpr)
+        .getStringValue()
+        .length() > 0
```

Comment:
This would now force that the host is a compile time constant. Previously it would have matched dynamic hosts values, e.g. returned by method calls, as well.
Maybe this could be solved by combining lines 109 to 128 to the following (formatting might need to be adjusted):
```ql
not exists (string hostString |
  hostString = host.(CompileTimeConstantExpr).getStringValue()
  or hostString = host
    .(VarAccess)
    .getVariable()
    .getAnAssignedValue()
    .(CompileTimeConstantExpr)
    .getStringValue()
|
    hostString.length() = 0 // Empty host is loopback address
    or hostString.regexpMatch(getPrivateHostRegex())
)
```

Reply:
Thanks @Marcono1234 a lot for the detailed instructions and I've updated the code. Now the code works without the unnecessary constraint.